From 71e5dba8964f558102622dc3f06a22d6f3c532d9 Mon Sep 17 00:00:00 2001
From: Omar Polo <op@omarpolo.com>
Date: Mon, 27 Jul 2020 10:21:24 +0200
Subject: [PATCH] don't mess with crowdin if the http client is disabled

---
 src/edframe.cpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/edframe.cpp b/src/edframe.cpp
index 2e8b46efe6..f2fa7db9d8 100644
--- a/src/edframe.cpp
+++ b/src/edframe.cpp
@@ -1775,9 +1775,14 @@ void PoeditFrame::OnUpdateFromSources(wxCommandEvent&)
 
 void PoeditFrame::OnUpdateFromSourcesUpdate(wxUpdateUIEvent& event)
 {
+#if HAVE_HTTP_CLIENT
     event.Enable(m_catalog &&
                  m_catalog->HasSourcesConfigured() &&
                  !CanSyncWithCrowdin(m_catalog));
+#else
+    event.Enable(m_catalog &&
+                 m_catalog->HasSourcesConfigured());
+#endif
 }
 
 void PoeditFrame::OnUpdateFromPOT(wxCommandEvent&)
@@ -2539,8 +2544,13 @@ void PoeditFrame::WarnAboutLanguageIssues()
         }
         else // no error, check for warning-worthy stuff
         {
-            if (lang.IsValid() && plForms != lang.DefaultPluralFormsExpr() && !CanSyncWithCrowdin(m_catalog))
+            if (lang.IsValid() && plForms != lang.DefaultPluralFormsExpr())
             {
+#ifdef HAVE_HTTP_CLIENT
+                if (!CanSyncWithCrowdin(m_catalog))
+                    return;
+#endif
+
                 AttentionMessage msg
                     (
                         "unusual-plural-forms",
